sources/host-ctr/cmd/host-ctr/main.go

```diff
@@ -38,6 +39,40 @@ import (
 // Example 2: 777777777777.dkr.ecr.cn-north-1.amazonaws.com.cn/my_image:latest
 var ecrRegex = regexp.MustCompile(`(^[a-zA-Z0-9][a-zA-Z0-9-_]*)\.dkr\.ecr\.([a-zA-Z0-9][a-zA-Z0-9-_]*)\.amazonaws\.com(\.cn)?.*`)
 
+// Used to define valid container types
+type ContainerType string
+
+const (
+	Host      ContainerType = "host"
+	Bootstrap ContainerType = "bootstrap"
+)
+
+// Returns an error if an invalid type is used
+func (ct ContainerType) IsValid() error {
+	switch ct {
+	case Host, Bootstrap:
+		return nil
+	}
+
+	return errors.New("Invalid container type")
+}
```

Comment:
In Go, it's typical to use a `bool` to indicate simple yes/no error states.  For example, when reading something that may or may not exist from a map, there's a `bool` returned and convention is to call it `ok`:

```go
val, ok := myMap[key]
```

For public methods, it's also [conventional](https://github.com/golang/go/wiki/CodeReviewComments#comment-sentences) for comments documenting declarations to begin with the name of the identifier.

```suggestion
// IsValid returns true if the container type is valid
func (ct ContainerType) IsValid() bool {
	switch ct {
	case Host, Bootstrap:
		return true
	}
	return false
}
```